src/sage/schemes/elliptic_curves/ell_point.py

```diff
@@ -308,6 +308,7 @@ def __init__(self, curve, v, check=True):
                 raise TypeError("Coordinates %s do not define a point on %s" % (list(v), curve))
 
         SchemeMorphism_point_abelian_variety_field.__init__(self, point_homset, v, check=False)
+        self._zxy_coords = (self._coords[2], self._coords[0], self._coords[1])
```

Comment:
Creating a second copy of all the coordinates seems a bit wasteful. I'd suggest constructing this tuple on the fly in `._richcmp_()`.